File: go/567_PermutationInString_test.go
package main

import (
    "github.com/stretchr/testify/assert"
    "testing"
)

func TestCheckInclusion(t *testing.T) {
    assert.Equal(t,
        true,
        checkInclusion("abb", "eidbbaooo"),
    )
    assert.Equal(t,
        true,
        checkInclusion("ab", "eidbaooo"),
    )
    assert.Equal(t,
        false,
        checkInclusion("ab", "eidboaoo"),
    )
    assert.Equal(t,
        false,
        checkInclusion("abb", "eidbaoo"),
    )
    assert.Equal(t,
        false,
        checkInclusion("hello", "ooolleoooleh"),
    )
    assert.Equal(t,
        true,
        checkInclusion("ab", "ab"),
    )
}

File: python/567_PermutationInString_test.py
import unittest


class TestSolution(unittest.TestCase):

    def test_checkInclusion(self):
        Solution = __import__('567_PermutationInString').Solution
        self.assertEqual(
            True,
            Solution().checkInclusion("abb", "eidbbaooo")
        )
        self.assertEqual(
            True,
            Solution().checkInclusion("ab", "eidbaooo")
        )
        self.assertEqual(
            False,
            Solution().checkInclusion("ab", "eidboaoo")
        )
        self.assertEqual(
            False,
            Solution().checkInclusion("abb", "eidbaoo")
        )
        self.assertEqual(
            False,
            Solution().checkInclusion("hello", "ooolleoooleh")
        )
        self.assertEqual(
            True,
            Solution().checkInclusion("ab", "ab")
        )
